packages/api/engine/opensea.go

```diff
@@ -72,38 +72,74 @@ func (o *Opensea) Sync(ctx context.Context) {
 
 				for _, asset := range ret.Assets {
 					var amount *big.Int
+					onSale := false
 					if asset.SellOrders != nil {
 						amount = asset.SellOrders[0].CurrentPrice.Big()
 						fmt.Println(asset.SellOrders[0].CurrentPrice)
+						onSale = true
 					} else {
 						amount = big.NewInt(0)
 					}
 
 					if err := tx.Asset.
 						Create().
-						SetID(fmt.Sprintf("%s", asset.ID)).
+						SetID(fmt.Sprintf("%d", asset.ID)).
 						SetAddress(string(asset.AssetContract.Address)).
 						SetSymbol(asset.AssetContract.Symbol).SetType("ETH").
 						SetAmount(schema.BigInt{Int: amount}).
+						SetDecimals(18).
 						OnConflictColumns("id").
-						Update(func(a *ent.AssetUpsert) {
-							a.AddAmount(schema.BigInt{Int: amount})
-						}).
+						UpdateNewValues().
 						Exec(ctx); err != nil {
 						fmt.Errorf("Error upserting to asset: %w", err)
+						return err
```

Comment:
lets return the formatted error from the line above. it can also be simplified to `upserting to asset: %w`